gwsumm/archive.py

```diff
@@ -178,8 +178,24 @@ def read_data_archive(sourcefile):
     ----------
     sourcefile : `str`
         path to source HDF5 file
+
+    rm_source_on_fail : `bool`, optional
+        remove the source HDF5 file if there was an OSError when opening the
+        file
     """
-    from h5py import File
+
+    # Check that the HDF5 source file is able to be opened.
+    # A common failure mode is that the HDF5 file is corrupted and this brings
+    # down the whole workflow, requiring manual intervention. Here, we attempt
+    # to automatically catch a common failure
+    try:
+        h5file = File(sourcefile, 'r')
+    except OSError as exc:  # file is corrupt, so we remove it to start fresh
```

Comment:
I think this needs to be a little more nuanced. If I naively try this with a file that doesn't exist:

```pytb
$ python -c "from gwsumm.archive import read_data_archive; print(read_data_archive('blah.h5'))"
/home/duncan/git/gwsumm/gwsumm/archive.py:196: UserWarning: failed to read blah.h5 [[Errno 2] Unable to open file (unable to open file: name = 'blah.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)], removing...
  warnings.warn(f"failed to read {sourcefile} [{exc}], removing...")
Traceback (most recent call last):
  File "/home/duncan/git/gwsumm/gwsumm/archive.py", line 192, in read_data_archive
    h5file = File(sourcefile, 'r')
  File "/home/duncan/opt/mambaforge/envs/py310/lib/python3.10/site-packages/h5py/_hl/files.py", line 533, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, fcpl, swmr=swmr)
  File "/home/duncan/opt/mambaforge/envs/py310/lib/python3.10/site-packages/h5py/_hl/files.py", line 226, in make_fid
    fid = h5f.open(name, flags, fapl=fapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5f.pyx", line 106, in h5py.h5f.open
FileNotFoundError: [Errno 2] Unable to open file (unable to open file: name = 'blah.h5', errno = 2, error message = 'No such file or directory', flags = 0, o_flags = 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/duncan/git/gwsumm/gwsumm/archive.py", line 197, in read_data_archive
    os.remove(sourcefile)
FileNotFoundError: [Errno 2] No such file or directory: 'blah.h5'
```

For simplicity I recommend manually catching `FileNotFoundError` and raising:

```suggestion
    except FileNotFoundError:
        raise
    except OSError as exc:  # file is corrupt, so we remove it to start fresh
```

Reply:
Done as suggested